service/service.go

```diff
@@ -53,7 +53,7 @@ func Run(ctx context.Context, cfg *config.Config, serviceList *ExternalServiceLi
 
 	// The proxy needs to be set up after the HealthCheck route has been added to the router: in the Setup method, the
 	// proxy adds a catch-all route, so any other routes added after that one will never be reachable.
-	p := proxy.Setup(ctx, r, cfg.BabbageURL)
+	p := proxy.Setup(ctx, r, *cfg)
```

Comment:
```suggestion
	p := proxy.Setup(ctx, r, cfg)
```
we customarily pass the pointer itself, so each function will need to be `func ...(..., cfg *config.Config, ...)`

Reply:
I actually did this on purpose. I didn't pass a pointer because I wanted to make clear that the configuration is not modified from this point on. It is also my understanding that the compiler is smart enough to optimise this code at compile time, and since this struct is relatively small, there shouldn't be a performance impact when copying the struct (or that's what I was hoping, at least).

Having said that, given that this is a proxy, performance does matter, so I'm happy to change it if necessary, but could you explain why passing a pointer would be the right thing to do, please? I'm new to Go, and I'd like to learn.

I think that in this instance, in both cases (pass by value vs pass by reference) the config is going to end up in the stack rather than in the heap (even when passing by reference, as we're just passing it down, never up), so providing the copy of the struct is very fast and not noticeable performance-wise (which, admittedly, I don't know if that's the case), then I chose to pass by value because it helps to convey the fact that it is only needed for reading purposes. But again, I'm happy to be corrected 👍

Reply:
All those points may indeed be valid, but some of our apps have very large configs and (even if the compiler optimises) it's still _in-house style_ to use a pointer, so I think I'll use that as my trump card.